src/main/java/at/hannibal2/skyhanni/features/garden/GardenCustomKeybinds.kt

```diff
@@ -0,0 +1,202 @@
+package at.hannibal2.skyhanni.features.garden
+
+import at.hannibal2.skyhanni.config.ConfigUpdaterMigrator
+import at.hannibal2.skyhanni.features.garden.CropType
+import at.hannibal2.skyhanni.features.garden.GardenAPI
+import at.hannibal2.skyhanni.mixins.transformers.AccessorKeyBinding
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.KeyboardManager.isKeyHeld
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiEditSign
+import net.minecraft.client.settings.KeyBinding
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import org.lwjgl.input.Keyboard
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable
+import java.util.IdentityHashMap
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object GardenCustomKeybinds {
+
+    private val config get() = GardenAPI.config.keyBind
+    private val mcSettings get() = Minecraft.getMinecraft().gameSettings
+
+    private val map: MutableMap<KeyBinding, MutableMap<String, () -> Int>> = IdentityHashMap()
+    private var currentCropName: String = ""
+    private var lastWindowOpenTime = SimpleTimeMark.farPast()
+    private var lastDuplicateKeybindsWarnTime = SimpleTimeMark.farPast()
+
+    init {
```

Comment:
A very long list like this was the reason we didn't want to create unique keybinds for every crop type in the first place

Reply:
I don't like the solution either, isn't there a way to make it without such a long list?

Reply:
to reduce code, we could reuse the same config object for every crop. but then we need to handle default values and the crop name in the description special.
also this would still not avoid the biggest problem: a very large config for the user, with tons to settings they might have changed accidentally

Reply:
I don't understand what the problem for the user would be exactly?